pkg/remote/client.go

```diff
@@ -0,0 +1,113 @@
+package remote
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	prompb "go.buf.build/grpc/go/prometheus/prometheus"
+)
+
+type HTTPConfig struct {
+	Timeout   time.Duration
+	TLSConfig *tls.Config
+	BasicAuth *BasicAuth
+	Headers   http.Header
+}
+
+type BasicAuth struct {
+	User, Password string
```

Comment:
non-blocking-nit: if we rename config ones to `User` it is also good to keep that synced

Reply:
Added, good catch, thanks

Reply:
I renamed to `Username`